### PR TITLE
Upgrade default TiDB version to v3.0.4

### PR DESCRIPTION
**Pull request by shonge**
### What problem does this PR solve? 
upgrade default TiDB version to v3.0.4
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test

Code changes

 - Has Helm charts change
 - Has Go code change

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
ACTION REQUIRED: upgrade default TiDB version to v3.0.4
 ```

### CODE REVIEW

File: tests/config.go [hunk 1]
```diff
@@ -78,7 +78,7 @@ func NewConfig() (*Config, error) {
 	flag.StringVar(&cfg.configFile, "config", "", "Config file")
 	flag.StringVar(&cfg.LogDir, "log-dir", "/logDir", "log directory")
 	flag.IntVar(&cfg.FaultTriggerPort, "fault-trigger-port", 23332, "the http port of fault trigger service")
-	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v3.0.0-rc.1,v3.0.0-rc.2,v3.0.1", "tidb versions")
+	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v3.0.1,v3.0.2", "tidb versions")
```
Comment (weekface) [inline tests/config.go]:
```suggestion
	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v3.0.0,v3.0.1,v3.0.2", "tidb versions")
```

**Comment by cofyc** [comment]
3.0.3 is released (https://github.com/pingcap/tidb/releases/tag/v3.0.3), can you update it to use 3.0.3? Thanks!

**Comment by cofyc** [comment]
some conflicts must be resolved

File: tests/config.go [hunk 2]
```diff
@@ -83,7 +83,7 @@ func NewConfig() (*Config, error) {
 	flag.StringVar(&cfg.configFile, "config", "", "Config file")
 	flag.StringVar(&cfg.LogDir, "log-dir", "/logDir", "log directory")
 	flag.IntVar(&cfg.FaultTriggerPort, "fault-trigger-port", 23332, "the http port of fault trigger service")
-	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v3.0.0,v3.0.1,v3.0.2", "tidb versions")
+	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v3.0.0,v3.0.1,v3.0.2,v3.0.3,v3.0.4", "tidb versions")
```
Comment (weekface) [inline tests/config.go]:
```suggestion
	flag.StringVar(&cfg.TidbVersions, "tidb-versions", "v3.0.2,v3.0.3,v3.0.4", "tidb versions")
```

The stability test used only the first three versions.

**Comment by tennix** [comment]
tidb-monitor-initializer v3.0.4 is not available yet, @qiffang please upgrade the tidb-monitor-initializer to the latest release version v3.0.4.

File: charts/tidb-cluster/values.yaml [hunk 3]
```diff
@@ -397,8 +397,8 @@ monitor:
   storageClassName: local-storage
   storage: 10Gi
   initializer:
-    image: pingcap/tidb-monitor-initializer:v3.0.1
-    imagePullPolicy: Always
+    image: pingcap/tidb-monitor-initializer:v3.0.4
```
Comment (weekface) [inline charts/tidb-cluster/values.yaml]: @qiffang there is no pingcap/tidb-monitor-initializer:v3.0.4 yet.  PTAL

File: tests/manifests/e2e/e2e.yaml [hunk 4]
```diff
@@ -48,7 +48,7 @@ spec:
     - /usr/local/bin/e2e
     - --operator-tag=e2e
     - --operator-image=pingcap/tidb-operator:latest
-    - --tidb-versions=v3.0.0-beta.1,v3.0.1
+    - --tidb-versions=v3.0.1,v3.0.2,v3.0.3,v3.0.4
```
Comment (weekface) [inline tests/manifests/e2e/e2e.yaml]:
```suggestion
    - --tidb-versions=v3.0.2,v3.0.3,v3.0.4
```

**Pull request merged**